### PR TITLE
WIP: config code clean

### PR DESCRIPTION
**Pull request by mrhardikjoshi**

### CODE REVIEW

File: ruby/neo4j/driver/config.rb [hunk 1]
```diff
@@ -42,39 +66,12 @@ def java_config
...
-      end
-
       private
 
       def init_security_and_trust_config(config)
-        trust_strategy = TrustStrategy.new(config)
+        trust_strategy = config[:trust_strategy] ? TrustStrategy.new(config) : DEFAULTS[:trust_strategy]
         merge!(
-          security_settings: Neo4j::Driver::Internal::SecuritySetting.new(config[:encrypted], trust_strategy),
+          security_settings: Neo4j::Driver::Internal::SecuritySetting.new(config[:encryption], trust_strategy),
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/config.rb:74]: the internal variable name in java-driver is `encrypted` but we have it as a `encryption` in ruby-driver so for backward compatibility I am keeping it as `encryption`

Comment (klobuczek) [reply ruby/neo4j/driver/config.rb:74]: yes, correct the API name in java is `withEncryption` and we derive all config keys from those method names

File: ruby/neo4j/driver/config.rb [hunk 2]
```diff
@@ -31,9 +36,28 @@ def cert_file_to_java
...
+        logger: ActiveSupport::Logger.new(STDOUT, level: ::Logger::ERROR), # :set_log
+        leaked_session_logging: false,
+        # connection_liveness_check_timeout: -1, # Not configured
+        max_connection_lifetime: 1.hour, # :set_max_connection_life_time
+        max_connection_pool_size: 100, #:set_max_pool_size
+        max_transaction_retry_time: Internal::Retry::ExponentialBackoffRetryLogic::DEFAULT_MAX_RETRY_TIME,
+        metrics_enabled: false,
+        # resolver: nil # :set_address_resolver
+        trust_strategy: TrustStrategy.trust_all_certificates,
+        user_agent: "neo4j-java/#{Neo4j::Driver::VERSION}"
```
Comment (klobuczek) [inline ruby/neo4j/driver/config.rb:55]: shouldn't it be `neo4j-ruby`?

File: ruby/neo4j/driver/config.rb [hunk 3]
```diff
@@ -22,6 +23,10 @@ def self.trust_all_certificates
           new(trust_strategy: TRUST_ALL_CERTIFICATES)
         end
 
+        def default_trust_strategy?
+          strategy == TRUST_ALL_CERTIFICATES
+        end
```
Comment (klobuczek) [inline ruby/neo4j/driver/config.rb:28]: what is this for?

**Review by klobuczek — APPROVED** [review]

**Pull request merged**